### DISCUSSION TITLE
Am I running into performance limits?

### DISCUSSION THREAD

**Discussion by AutomaRobin** [Budibase/budibase#8951] [Help] [answered]
Hi, I'm self-hosting Budibase and it's been a smooth ride until today. I am running into an issue where the Budibase back-end gives 503 Service Unavailable errors.
My setup is a dashboard where I can fill in EAN codes of which I'll track the supply at one of the big retail platforms on The Netherlands.

I currently have 19 products which I track. Each of those products are mapped to a table in which you can see all suppliers and their given supply at an hour of the day. Each table has its own date selector.
![image](https://user-images.githubusercontent.com/26344173/205938524-207ac6f3-d2e6-47e4-b8a1-93c1cda800a5.png)

The problem started today when I added a few new products. Now each time I open this page in my dashboard I see a list "Service unavailable" errors and the same number of tables that are empty.

![image](https://user-images.githubusercontent.com/26344173/205938184-89f38721-4d5c-4324-b95c-f48685203515.png)

When I check the Network tab, I do see that all tables are loaded without the current date and the query falls back on the default binding value. After, I think, the date selectors are loaded, the data source is requested again but now with the current date. But this second time, the requests start to fail and the data tables end up empty.

![image](https://user-images.githubusercontent.com/26344173/205939468-24bb7a04-04bf-4d34-92b8-26b890afcda3.png)

To give some context, I currently do a query to select all EAN codes of the products I like to track. I loop over this to be able to use each EAN as input for the supply per hour query. This query takes a date and EAN code as input and will retrieve the data shown in the first screenshot. No problems arise when I limit the EAN query to only show 10 products. So it definitely seems like it's related to the number of requests that are executed within a few seconds.

Anyone that has more insights into my issue?

**Comment by shogunpurple** [comment, marked as answer]
Hi @AutomaRobin  - looks like you are potentially being rate limited here.

In your `docker-compose.yaml` file, there's a variable there to update your rate limit in the NGINX proxy. Try updating this to a higher number and restarting your docker containers.

```
PROXY_RATE_LIMIT_API_PER_SECOND=100
```

**Reply by AutomaRobin** [reply]
Does this also apply for the single service image `budibase/budibase:latest`?

**Reply by AutomaRobin** [reply]
@shogunpurple?

**Reply by shogunpurple** [reply]
@AutomaRobin ah it doesn’t actually. This is docker compose only at the moment. I’ll convert this into a ticket that enables support for this on the single image. In the meantime you could edit the nginx.conf directly from a docker bind mount and remove the rate limit.

**Reply by shogunpurple** [reply]
https://github.com/Budibase/budibase/issues/8978

**Reply by AutomaRobin** [reply]
@shogunpurple Do you know where I can find the nginx.conf file in the docker volume? I'm currently in the _data folder and can find the `.env`, `couch`, `minio` and `search` folders. But nothing indicating the nginx configuration file in those folders. Also adding the `PROXY_RATE_LIMIT_API_PER_SECOND` in the `.env` file doesn't improve/fix my problem.

**Reply by shogunpurple** [reply]
`/etc/nginx.conf` is where you will find the config - the default path for nginx. You'll have to restart nginx after any config updates.

If you add anything to the `.env` you will need to restart the services, did you do that?

**Reply by AutomaRobin** [reply]
@shogunpurple I guess that does not apply to the single image, right? As the binded volume does not have an /etc/ folder nor any nginx.conf files. And yes, I have restarted the services after adding he environment variables.

**Reply by shogunpurple** [reply]
It does, if you run:
`docker exec <your container id> cat /etc/nginx/nginx.conf` you can see the config. If you want to use your own custom one, you will need to set up your own bind mount to point to this path on the container and edit the file.

**Reply by AutomaRobin** [reply]
@shogunpurple That worked! Thanks for your help
